lib/rucio/core/rse_counter.py

```diff
@@ -128,15 +130,70 @@ def update_rse_counter(rse_id, session=None):
     :param session:  Database session in use.
     """
 
-    updated_rse_counters = session.query(models.UpdatedRSECounter).\
-        filter_by(rse_id=rse_id).all()
+    updated_rse_counters = session.query(models.UpdatedRSECounter).filter_by(rse_id=rse_id).all()
+    sum_bytes = sum([updated_rse_counter.bytes for updated_rse_counter in updated_rse_counters])
+    sum_files = sum([updated_rse_counter.files for updated_rse_counter in updated_rse_counters])
 
     try:
         rse_counter = session.query(models.RSEUsage).filter_by(rse_id=rse_id, source='rucio').one()
-        rse_counter.used += sum([updated_rse_counter.bytes for updated_rse_counter in updated_rse_counters])
-        rse_counter.files += sum([updated_rse_counter.files for updated_rse_counter in updated_rse_counters])
+        rse_counter.used += sum_bytes
+        rse_counter.files += sum_files
     except NoResultFound:
-        pass
+        models.RSEUsage(rse_id=rse_id,
+                        used=sum_bytes,
+                        files=sum_files,
+                        source='rucio').save(session=session)
 
     for update in updated_rse_counters:
         update.delete(flush=False, session=session)
+
+
+@read_session
+def get_rse_usage_from_unavailable_replicas(total_workers, worker_number, session=None):
```

Comment:
I am not quite sure where this requirement comes from? I don't think that the PL/SQL procedure fills unavailable replica info?

Reply:
I thought it belongs to [this part of the procedure](https://github.com/rucio/rucio/blob/master/etc/sql/oracle/procedures.sql#L210)

Reply:
Ahh I see; Hmm the issue is that in the procedure this is really an optimised query going against a special index in Oracle. For big tables this could be a bit of a show-stopper, but let's leave it for now. 👍